src/jace/jax/stages.py

```diff
@@ -0,0 +1,325 @@
+        This function is used by the cache to generate the key.
+        """
+        fargs = tuple(tcache._AbstarctCallArgument.from_value(x) for x in args)
+        return tcache.CachedCallDescription(stage_id=id(self), fargs=fargs)
+
+
+class JaceLowered(Stage):
+    """Represents the original computation that was lowered to SDFG."""
+
+    DEF_COMPILER_OPTIONS: Final[dict[str, Any]] = {
+        "auto_optimize": True,
+        "simplify": True,
+    }
+
+    # `self` assumes complete ownership of the
+    _trans_sdfg: translator.TranslatedJaxprSDFG
+    _cache: tcache.TranslationCache
+
+    def __init__(
+        self,
+        trans_sdfg: translator.TranslatedJaxprSDFG,
+    ) -> None:
+        """Constructs the lowered object."""
+        if not trans_sdfg.is_finalized:
+            raise ValueError("The translated SDFG must be finalized.")
+        if trans_sdfg.inp_names is None:
+            raise ValueError("Input names must be defined.")
+        if trans_sdfg.out_names is None:
+            raise ValueError("Output names must be defined.")
+        self._trans_sdfg = trans_sdfg
+        self._cache = tcache.get_cache(self)
+
+    @tcache.cached_translation
+    def compile(
+        self,
+        compiler_options: CompilerOptions | None = None,
+    ) -> JaceCompiled:
+        """Compile the SDFG.
+
+        Returns an Object that encapsulates a compiled SDFG object.
+        You can pass a `dict` as argument which are passed to the `jace_optimize()` routine.
+        If you pass `None` then the default options are used.
+        To disable all optimization, pass an empty `dict`.
+
+        Notes:
+            I am pretty sure that `None` in Jax means "use the default option".
+                See also `CachedCallDescription.make_call_description()`.
+        """
+        # We **must** deepcopy before we do any optimization.
+        #  There are many reasons for this but here are the most important ones:
+        #  All optimization DaCe functions works in place, if we would not copy the SDFG first, then we would have a problem.
+        #  Because, these optimization would then have a feedback of the SDFG object which is stored inside `self`.
+        #  Thus, if we would run this code `(jaceLoweredObject := jaceWrappedObject.lower()).compile({opti=True})` would return
+        #  an optimized object, which is what we intent to do.
+        #  However, if we would now call `jaceWrappedObject.lower()` (with the same arguments as before), we should get `jaceLoweredObject`,
+        #  since it was cached, but it would actually contain an already optimized SDFG, which is not what we want.
+        #  If you think you can remove this line then do it and run `tests/test_decorator.py::test_decorator_sharing`.
+        fsdfg: translator.TranslatedJaxprSDFG = copy.deepcopy(self._trans_sdfg)
+        optimization.jace_optimize(
+            fsdfg, **(self.DEF_COMPILER_OPTIONS if compiler_options is None else compiler_options)
+        )
```

Comment:
I am not sure, but I want that `lowered.compile()` uses the default options and `lowered.compile({})` compiles uses all options disabled.
Because, this is what `jace_optimize()` does.

I agree that the call `lowered.compile(None)` means "use default is not that good".